transaction.go

```diff
@@ -722,23 +722,31 @@ func TxSemanticNativeTokens() TxSemanticValidationFunc {
 	return func(svCtx *SemanticValidationContext) error {
 		// native token set creates handle overflows
 		var err error
-		var inNTCount, outNTCount int
-		svCtx.WorkingSet.InNativeTokens, inNTCount, err = svCtx.WorkingSet.Inputs.NativeTokenSum()
+		svCtx.WorkingSet.InNativeTokens, err = svCtx.WorkingSet.Inputs.NativeTokenSum()
 		if err != nil {
 			return fmt.Errorf("invalid input native token set: %w", err)
 		}
+		inNTCount := len(svCtx.WorkingSet.InNativeTokens)
```

Comment:
Nature loves symmetry, so let's have the same check for `outNTCount` before calculating `distinctNativeCount`.

Reply:
The transaction would not pass syntactical validation if the count would exceed the distinct native count beforehand.

Reply:
Yes, that's clear. I wast just suggesting this optimization in order not to compute `distinctNTCount` when `len(svCtx.WorkingSet.OutNativeTokens)` would already exceed the limit.

Reply:
I do not understand the optimisation: `len(svCtx.WorkingSet.OutNativeTokens)` can not exceed `Max Native Tokens Count` here as the transaction would already fail at syntactical validation otherwise, therefore the check will never be true.

On the other hand for inputs this check makes sense as the inputs are only available at semantical validation.

Reply:
Good point. I just realized it is indeed checked in https://github.com/iotaledger/iota.go/blob/2f37dd013dee53077201133ae74aa67559cdfdc0/output.go#L748 Let me add this check to TIP-20 because it is not present in `Transaction Syntactical Validation Rules`. According to TIP-20, native token counts are checked per output (syntactic) and then in the semantic tx validation accross the all transaction, but not for only outputs.